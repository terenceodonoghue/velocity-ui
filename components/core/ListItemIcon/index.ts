export { default } from './ListItemIcon';
export type { ListItemIconProps } from './ListItemIcon.types';
