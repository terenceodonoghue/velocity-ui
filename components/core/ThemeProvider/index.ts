export { default, defaultTheme } from './ThemeProvider';
export type { Theme } from './ThemeProvider.types';
