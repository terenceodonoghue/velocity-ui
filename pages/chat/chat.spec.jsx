/** @jsx jsx */
import { jsx } from '@emotion/core';
import faker from 'faker';
import moment from 'moment';
import { Conversations } from './index';

describe('Chat', () => {
  describe('Conversations', () => {
    it('matches snapshot', () => {
      const tree = renderer
        .create(
          withTheme(
            <Conversations
              data={[
                {
                  isOnline: faker.random.boolean(),
                  message: faker.lorem.sentences(),
                  name: faker.name.findName(),
                  src: faker.image.avatar(),
                  time: moment()
                    .minutes(faker.random.number({ min: 1, max: 19 }))
                    .format('m[m]'),
                },
                {
                  isOnline: faker.random.boolean(),
                  message: faker.lorem.sentences(),
                  name: faker.name.findName(),
                  src: faker.image.avatar(),
                  time: moment()
                    .minutes(faker.random.number({ min: 20, max: 39 }))
                    .format('m[m]'),
                },
                {
                  isOnline: faker.random.boolean(),
                  message: faker.lorem.sentences(),
                  name: faker.name.findName(),
                  src: faker.image.avatar(),
                  time: moment()
                    .minutes(faker.random.number({ min: 40, max: 59 }))
                    .format('m[m]'),
                },
                {
                  isOnline: faker.random.boolean(),
                  message: faker.lorem.sentences(),
                  name: faker.name.findName(),
                  src: faker.image.avatar(),
                  time: moment()
                    .hours(faker.random.number({ min: 1, max: 5 }))
                    .format('h[h]'),
                },
                {
                  isOnline: faker.random.boolean(),
                  message: faker.lorem.sentences(),
                  name: faker.name.findName(),
                  src: faker.image.avatar(),
                  time: moment()
                    .hours(faker.random.number({ min: 6, max: 11 }))
                    .format('h[h]'),
                },
                {
                  isOnline: faker.random.boolean(),
                  message: faker.lorem.sentences(),
                  name: faker.name.findName(),
                  src: faker.image.avatar(),
                  time: moment()
                    .hours(faker.random.number({ min: 12, max: 17 }))
                    .format('h[h]'),
                },
                {
                  isOnline: faker.random.boolean(),
                  message: faker.lorem.sentences(),
                  name: faker.name.findName(),
                  src: faker.image.avatar(),
                  time: moment()
                    .hours(faker.random.number({ min: 18, max: 23 }))
                    .format('h[h]'),
                },
              ]}
            />,
          ),
        )
        .toJSON();

      expect(tree).toMatchSnapshot();
    });
  });
});
