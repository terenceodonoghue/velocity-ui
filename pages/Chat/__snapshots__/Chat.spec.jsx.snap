// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Chat Conversations matches snapshot 1`] = `
.emotion-56 {
  border-radius: 1px;
  height: 100%;
  margin: 0;
  padding: 0;
  width: 378px;
  background-color: #fff;
  list-style: none;
  overflow: hidden;
}

.emotion-7 {
  display: block;
  border-radius: 1px;
  border: solid 1px rgba(46,91,255,0.08);
  box-shadow: 0 10px 20px 0 rgba(46,91,255,0.07);
  padding: 32px 24px;
  background-color: #fff;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  border-left: 0;
  border-top: 0;
  padding: 17px 25px;
  cursor: pointer;
  -webkit-transition: all 0.2s linear;
  transition: all 0.2s linear;
}

.emotion-7:hover {
  background-color: rgba(46,91,255,0.08);
}

.emotion-1 {
  -webkit-box-flex: 0;
  -webkit-flex-grow: 0;
  -ms-flex-positive: 0;
  flex-grow: 0;
  margin-right: 10px;
}

.emotion-0 {
  border-radius: 50%;
}

.emotion-6 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
}

.emotion-4 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  line-height: 1.47;
}

.emotion-2 {
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
  font-weight: 500;
}

.emotion-3 {
  color: #8798ad;
}

.emotion-5 {
  display: block;
  color: #8798ad;
  line-height: 1.47;
  max-width: 255px;
  white-space: nowrap;
  overflow: hidden;
  text-overflow: ellipsis;
}

<ul
  className="emotion-56"
>
  <div
    className="emotion-7"
  >
    <div
      className="emotion-1"
    >
      <img
        className="emotion-0"
        height="48"
        src="https://s3.amazonaws.com/uifaces/faces/twitter/barputro/128.jpg"
        width="48"
      />
    </div>
    <div
      className="emotion-6"
    >
      <div
        className="emotion-4"
      >
        <span
          className="emotion-2"
        >
          Roslyn Kulas
        </span>
        <span
          className="emotion-3"
        >
          15m
        </span>
      </div>
      <span
        className="emotion-5"
      >
        Molestiae tempora eligendi omnis quisquam. Quos nihil dolor voluptatibus velit nobis culpa deleniti. Reprehenderit in nisi et. Quia odio et inventore eligendi in deserunt id. Sit odio quia vitae provident quo provident molestiae.
      </span>
    </div>
  </div>
  <div
    className="emotion-7"
  >
    <div
      className="emotion-1"
    >
      <img
        className="emotion-0"
        height="48"
        src="https://s3.amazonaws.com/uifaces/faces/twitter/nitinhayaran/128.jpg"
        width="48"
      />
    </div>
    <div
      className="emotion-6"
    >
      <div
        className="emotion-4"
      >
        <span
          className="emotion-2"
        >
          Johnpaul Kutch
        </span>
        <span
          className="emotion-3"
        >
          32m
        </span>
      </div>
      <span
        className="emotion-5"
      >
        Esse quia expedita commodi autem tempora ut exercitationem. Harum voluptas sunt omnis et accusantium et quia quos sequi. Beatae enim necessitatibus molestiae placeat saepe.
      </span>
    </div>
  </div>
  <div
    className="emotion-7"
  >
    <div
      className="emotion-1"
    >
      <img
        className="emotion-0"
        height="48"
        src="https://s3.amazonaws.com/uifaces/faces/twitter/muringa/128.jpg"
        width="48"
      />
    </div>
    <div
      className="emotion-6"
    >
      <div
        className="emotion-4"
      >
        <span
          className="emotion-2"
        >
          Reyes Schmeler V
        </span>
        <span
          className="emotion-3"
        >
          50m
        </span>
      </div>
      <span
        className="emotion-5"
      >
        Voluptatem dolores esse nobis et. Magnam aut atque dolores maiores incidunt occaecati iusto laborum facere.
      </span>
    </div>
  </div>
  <div
    className="emotion-7"
  >
    <div
      className="emotion-1"
    >
      <img
        className="emotion-0"
        height="48"
        src="https://s3.amazonaws.com/uifaces/faces/twitter/jitachi/128.jpg"
        width="48"
      />
    </div>
    <div
      className="emotion-6"
    >
      <div
        className="emotion-4"
      >
        <span
          className="emotion-2"
        >
          Mrs. Norene Toy
        </span>
        <span
          className="emotion-3"
        >
          1h
        </span>
      </div>
      <span
        className="emotion-5"
      >
        Corporis laboriosam esse. Soluta accusamus officiis ut excepturi blanditiis libero ut. Quo fuga enim tempore non voluptas quia.
      </span>
    </div>
  </div>
  <div
    className="emotion-7"
  >
    <div
      className="emotion-1"
    >
      <img
        className="emotion-0"
        height="48"
        src="https://s3.amazonaws.com/uifaces/faces/twitter/0therplanet/128.jpg"
        width="48"
      />
    </div>
    <div
      className="emotion-6"
    >
      <div
        className="emotion-4"
      >
        <span
          className="emotion-2"
        >
          Merle Hermiston
        </span>
        <span
          className="emotion-3"
        >
          6h
        </span>
      </div>
      <span
        className="emotion-5"
      >
        Explicabo occaecati rerum nesciunt. Sint est asperiores sit voluptatum mollitia enim iste. Nesciunt minima sequi voluptas optio aut voluptatem. Eligendi voluptates iste eius iure commodi molestiae.
      </span>
    </div>
  </div>
  <div
    className="emotion-7"
  >
    <div
      className="emotion-1"
    >
      <img
        className="emotion-0"
        height="48"
        src="https://s3.amazonaws.com/uifaces/faces/twitter/_kkga/128.jpg"
        width="48"
      />
    </div>
    <div
      className="emotion-6"
    >
      <div
        className="emotion-4"
      >
        <span
          className="emotion-2"
        >
          Aletha O'Reilly
        </span>
        <span
          className="emotion-3"
        >
          12h
        </span>
      </div>
      <span
        className="emotion-5"
      >
        In unde nihil iste asperiores consequatur. Quidem omnis inventore deserunt ipsa. Officiis voluptatem nostrum in omnis labore consequatur nisi.
      </span>
    </div>
  </div>
  <div
    className="emotion-7"
  >
    <div
      className="emotion-1"
    >
      <img
        className="emotion-0"
        height="48"
        src="https://s3.amazonaws.com/uifaces/faces/twitter/justme_timothyg/128.jpg"
        width="48"
      />
    </div>
    <div
      className="emotion-6"
    >
      <div
        className="emotion-4"
      >
        <span
          className="emotion-2"
        >
          Andreanne Hauck
        </span>
        <span
          className="emotion-3"
        >
          9h
        </span>
      </div>
      <span
        className="emotion-5"
      >
        Quia praesentium distinctio adipisci nulla exercitationem omnis illum at natus. Cum ipsam consequatur ex accusamus ipsum.
      </span>
    </div>
  </div>
</ul>
`;
